compiler/noirc_frontend/src/hir/comptime/errors.rs

```diff
@@ -96,12 +99,23 @@ impl InterpreterError {
             | InterpreterError::UnquoteFoundDuringEvaluation { location, .. }
             | InterpreterError::Unimplemented { location, .. }
             | InterpreterError::BreakNotInLoop { location, .. }
-            | InterpreterError::ContinueNotInLoop { location, .. } => *location,
+            | InterpreterError::ContinueNotInLoop { location, .. }
+            | InterpreterError::DebugEvaluateComptime { location, .. } => *location,
             InterpreterError::Break | InterpreterError::Continue => {
                 panic!("Tried to get the location of Break/Continue error!")
             }
         }
     }
+
+    pub(crate) fn debug_evaluate_comptime(expr: impl Display, location: Location) -> Self {
+        let displayed_expr: String = format!("{}", expr);
+        let diagnostic = CustomDiagnostic::simple_debug(
+            "`comptime` expression ran:".to_string(),
+            format!("After evaluation:\n{}", displayed_expr),
+            location.span,
+        );
```

Comment:
```suggestion
        let diagnostic = CustomDiagnostic::simple_debug(
            "`comptime` expression ran:".to_string(),
            format!("After evaluation:\n{}", expr),
            location.span,
        );
```

This should be equivalent.